packages/api/src/graphql/mutation.public.ts

```diff
@@ -79,6 +89,109 @@ export const GraphQLPublicMutation = new GraphQLObjectType<undefined, Context>({
         const session = authenticateUser()
         return session ? await dbAdapter.session.deleteUserSessionByToken(session.token) : false
       }
+    },
+
+    registerMemberAndReceivePayment: {
+      type: GraphQLNonNull(GraphQLPublicPayment),
+      args: {
+        name: {type: GraphQLNonNull(GraphQLString)},
+        preferredName: {type: GraphQLString},
+        email: {type: GraphQLNonNull(GraphQLString)},
+        memberPlanID: {type: GraphQLNonNull(GraphQLString)},
+        autoRenew: {type: GraphQLNonNull(GraphQLBoolean)},
+        paymentPeriodicity: {type: GraphQLNonNull(GraphQLPaymentPeriodicity)},
+        monthlyAmount: {type: GraphQLNonNull(GraphQLInt)},
+        paymentMethodID: {type: GraphQLNonNull(GraphQLString)},
+        successURL: {type: GraphQLString},
+        failureURL: {type: GraphQLString}
+      },
+      async resolve(
+        root,
+        {
+          name,
+          preferredName,
+          email,
+          memberPlanID,
+          autoRenew,
+          paymentPeriodicity,
+          monthlyAmount,
+          paymentMethodID,
+          successURL,
+          failureURL
+        },
+        {dbAdapter, loaders, authenticateUser, memberContext, createPaymentWithProvider}
+      ) {
+        /* const userSession = authenticateUser()
+        if(userSession.user) throw new Error('Can not register authenticated user') // TODO: check this again
+        */
+
+        const memberPlan = await loaders.activeMemberPlansByID.load(memberPlanID)
+        if (!memberPlan) throw new NotFound('MemberPlan', memberPlanID)
+
+        const paymentMethod = await loaders.activePaymentMethodsByID.load(paymentMethodID)
+        if (!paymentMethod) throw new NotFound('PaymentMethod', paymentMethodID)
+
+        if (monthlyAmount <= memberPlan.amountPerMonthMin) throw new MonthlyAmountNotEnough()
```

Comment:
Do you intend for monthlyAmount = memberPlan.amountPerMonthMin to throw an error? Should it not be: monthlyAmount < memberPlan.amountPerMonthMin ?

Reply:
You're absolutely right. Thanks @penk85 👍🏾